Makefile

```diff
@@ -150,6 +150,9 @@ olm-push:
 bump-kubevirtci:
 	./hack/bump-kubevirtci.sh
 
+parallel-sync:
```

Comment:
The name is misleading. It suggests it performs multiple syncs in parallel, but it does not.

Why do we need to keep this as an additional target? If it is not in the mainstream workflow, it will get broken and outdated. Does anything prevent this from being the cluster-sync?

Reply:
Roman suggested it wont be the main one, because he doesnt want to run parallel process on prow,
btw, since prow always create a cluster from scratch it wont run it parallel, because it will detect there is no cluster, so in this case we can use it and save around  1 - 1.5 minute or so.

another reason is that its mostly for local use, where you control the cluster and in case your cluster is messed up, you do want to force clean everything using the `make clean-cluster` that is part of `make cluster-sync`.
i can change the name
locally it will save time when redeploying and on first deployment, around 1 - 1.5 minute out of the ~4.5 when one develop and do `make cluster-sync` couple times in a day.

Reply:
`make cluster-sync` is meant for users, so maybe this should go under that target and Prow would use a different one? Or does this have any regression for local use?

Reply:
this one is more manual user oriented which control his cluster
both prow and locally we use make `cluster-sync` atm
as long as prow always runs from scratch (which i guess wont change) it is safe to change it to this target, but maybe less explicit

this target is more advance, in the meaning that for a broken cluster you might need to know what you are doing and run make cluster-clean or make sync, 99% of the time cluster doesnt break and this one is safe to use and saves time.

wdyt?

Reply:
So this new target may not be suitable for edge scenarios. I don't see that as a reason not to make it the default. You can change the target used by CI.

My worry is that if this does not become the default, it will have a single user at best.

Reply:
ok so you suggest to rename it and use it in CI?
we need to ask Roman about it, to make sure he accept, 

this is the only AI needed?

Reply:
If your new target does not introduce any regression when used by humans, I see no reason why it should not be the default `make cluster-sync`. To make CI more stable, the original target should be renamed or parametrized and applied on Prow.

If this introduces a regression, it should be spelled out.

In both cases, you should document this new target in development docs.

Reply:
> If your new target does not introduce any regression when used by humans, I see no reason why it should not be the default `make cluster-sync`. To make CI more stable, the original target should be renamed or parametrized and applied on Prow.
> 
> If this introduces a regression, it should be spelled out.
it does in case the cluster is broken, this is why we cant omit the other, on CI we can change to this one, but i am not sure Roman would like it, or if its mandatory beside what you did about no regression check.

> 
> In both cases, you should document this new target in development docs.

one thing important to note, i did see with good ole make cluster-sync few times already that the changes i did didnt catch in the cluster, so in case it happens to people who will use this method manually, its not for sure because of this method.

Reply:
> one thing important to note, i did see with good ole make cluster-sync few times already that the changes i did didnt catch in the cluster, so in case it happens to people who will use this method manually, its not for sure because of this method.

You say that the new target is not reliable even on a healthy cluster? Do we want to keep it in the project then?

Reply:
> > one thing important to note, i did see with good ole make cluster-sync few times already that the changes i did didnt catch in the cluster, so in case it happens to people who will use this method manually, its not for sure because of this method.
> 
> You say that the new target is not reliable even on a healthy cluster? Do we want to keep it in the project then?

i talk on the old target `make cluster-sync` , not on the new, just wanted to bring it to the table that there are already rare cases (and maybe as well other people saw them) on the current method, where some updates doesnt catch after sync,
sometimes i had to kill the virt-handler pods in order for it to catch after sync, sometimes i just did cluster-sync without checking alternatives methods. but the important thing is that i talk on the current method, not the new method, and wanted it to be known that if it happens its not for sure because the new target.

Reply:
> i talk on the old target make cluster-sync , not on the new, just wanted to bring it to the table that there are already rare cases (and maybe as well other people saw them) on the current method, where some updates doesnt catch after sync,

How does it help with the review of this PR?

Reply:
just wanted to add this info because it sounds important at least in my view. so it would be documented
disregard it please if you find it unreleated.

Reply:
What about the documentation of this new target? Are you going to add it to the project?

Reply:
of course, just trying to grasp first what we want to do beside if any,
looking on CI, this is the code right now, in my opinion i would suggest not to change it in this PR, but add this new target first with documentation and then see ?
it even doesn't use `make clluster-sync`

Note: `make cluster-deploy` does run `make cluster-clean`
`automation/test.sh`
```
# I do not have good indication that OKD API server ready to serve requests, so I will just
# repeat cluster-deploy until it succeeds
until make cluster-deploy; do
    sleep 1
done
```

Reply:
done, lets see all pass

Reply:
passed